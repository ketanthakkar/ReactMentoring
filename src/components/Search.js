import React, { Component } from 'react';
import { fetchMovies, filterMovies, filterMoviesBy } from '../actions/index';
import { bindActionCreators } from 'redux'; 
import { connect } from 'react-redux';

class Search extends Component {

    handleSearchClick = (event) => {
       const filterBy = event.target.id === "genre-btn" ? "genres" : "title"
       this.props.filterMoviesBy(filterBy);

       document.getElementById("title-btn").classList.toggle("nonselected-color");
       document.getElementById("genre-btn").classList.toggle("nonselected-color");
    }

    handleChange = (event) => {
        this.props.filterMovies(event.target.value);
      }

    render() {
        return (
            <section className="search-section">
                <h4 className="white-text">FIND YOUR</h4>
                <input type="text" placeholder="Enter title" className="white-text search-title"
                        onChange={this.handleChange}></input>
                <div className="search-container">
                    <div className="search-selection">
                        <label className="white-text">SEARCH BY</label>
                        <button id="title-btn" className="white-text" onClick={this.handleSearchClick}>TITLE</button>
                        <button id="genre-btn" className="white-text nonselected-color" onClick={this.handleSearchClick}>GENRE</button>
                    </div>
                    <button className="white-text" onClick={this.props.fetchMovies}>SEARCH</button>
                </div>
            </section>
            )
        }
    }

const mapDispatchToProps = (dispatch) => {
    return bindActionCreators({
        fetchMovies,
        filterMovies,
        filterMoviesBy
    }, dispatch)
};

export default connect(null, mapDispatchToProps)(Search);