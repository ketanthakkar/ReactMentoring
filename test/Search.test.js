import React from 'react';
import { mount } from 'enzyme';
import Search from "../src/components/Search";

describe('should render Search correctly', () => {
  const tree = mount(<Search />);
  test('Snapshot test', () => {
    expect(tree).toMatchSnapshot();
  });
});
