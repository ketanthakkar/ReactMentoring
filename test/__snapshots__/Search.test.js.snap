// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Search correctly Snapshot test 1`] = `
<Search>
  <section
    className="search-section"
  >
    <h4
      className="white-text"
    >
      FIND YOUR MOVIE
    </h4>
    <input
      className="white-text search-title"
      placeholder="Enter title"
      type="text"
    />
    <div
      className="search-container"
    >
      <div
        className="search-selection"
      >
        <label
          className="white-text"
        >
          SEARCH BY
        </label>
        <button
          className="white-text"
          id="title-btn"
        >
          TITLE
        </button>
        <button
          className="white-text"
          id="genre-btn"
        >
          GENRE
        </button>
      </div>
      <button
        className="white-text"
      >
        SEARCH
      </button>
    </div>
  </section>
</Search>
`;
